### PR TITLE
Update option to skip test in generators

### PR DESCRIPTION
**Pull request by mrgilman**
Rails no longer generates Test::Unit files by default.

### CODE REVIEW

File: railties/lib/rails/generators/app_base.rb [hunk 1]
```diff
@@ -69,8 +69,8 @@ def self.add_shared_options_for(name)
         class_option :skip_turbolinks,    type: :boolean, default: false,
                                           desc: 'Skip turbolinks gem'
 
-        class_option :skip_test_unit,     type: :boolean, aliases: '-T', default: false,
-                                          desc: 'Skip Test::Unit files'
+        class_option :skip_test,          type: :boolean, aliases: '-T', default: false,
```
Comment (rafaelfranca) [inline railties/lib/rails/generators/app_base.rb]: If we are renaming the option should not we rename its usage?

Comment (mrgilman) [inline railties/lib/rails/generators/app_base.rb]: Yup! Fixed.

**Comment by sgrif** [comment]
This needs a changelog entry.

**Comment by mrgilman** [comment]
Added a changelog entry.

File: railties/lib/rails/generators/app_base.rb [hunk 2]
```diff
@@ -69,8 +69,8 @@ def self.add_shared_options_for(name)
...
+                                          desc: 'Skip ActiveSupport::TestCase files'
```
Comment (carlosantoniodasilva) [inline railties/lib/rails/generators/app_base.rb]: I guess there's no need to be super explicit, maybe we could simply say `Skip test files`, wdyt?

Comment (rafaelfranca) [inline railties/lib/rails/generators/app_base.rb]: :+1:

Comment (mrgilman) [inline railties/lib/rails/generators/app_base.rb]: I agree, this sounds better. I'll update.

File: railties/test/generators/plugin_generator_test.rb [hunk 3]
```diff
@@ -333,14 +333,14 @@ def test_creating_dummy_without_tests_but_with_dummy_path
...
   def test_ensure_that_gitignore_can_be_generated_from_a_template_for_dummy_path
     FileUtils.cd(Rails.root)
-    run_generator([destination_root, "--dummy_path", "spec/dummy", "--skip-test-unit"])
+    run_generator([destination_root, "--dummy_path", "spec/dummy", "--skip-test"])
     assert_file ".gitignore" do |contents|
       assert_match(/spec\/dummy/, contents)
     end
   end
 
   def test_skipping_test_unit
```
Comment (carlosantoniodasilva) [inline railties/test/generators/plugin_generator_test.rb]: Maybe you could rename the test here too.

File: railties/test/generators/shared_generator_tests.rb [hunk 4]
```diff
@@ -48,7 +48,7 @@ def test_invalid_database_option_raises_an_error
   end
 
   def test_test_unit_is_skipped_if_required
```
Comment (carlosantoniodasilva) [inline railties/test/generators/shared_generator_tests.rb]: And here as well?

**Pull request merged**